prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -188,7 +188,7 @@ object FHIRBundleHelpers {
                 val value = it.getter.call(res.resource)
                 if (value is MutableList<*>) {
                     value.removeIf { it is Reference && it.reference == resource.idBase }
-                } else if (value is Reference) {
+                } else if (value is Reference && value.reference == resource.idBase) {
                     value.reference = null
```

Comment:
Not sure how I missed this. This was deleting references for every reference. We really need an integration test for this.

Reply:
Please create a new ticket and link in this PR.